perch/cli.py

```diff
@@ -0,0 +1,120 @@
+import json
+import click
+
+from perch import PerchAPIClient
+
+
+ROOT_URL = 'http://api.local.perchsecurity.com'
+
+COMPANY_ID = 1
+
+COMMUNITIES = (
+    '16', #MPSISAO dev
+    '29', #MPSISAO
+)
+
+COLUMNS = {
+    'id': 0,
+    'title': 6,
+    'tlp': 7,
+    'description': 3,
+    'confidence': 9,
+    'observable_type': 5,
+    'observable_value': 1,
+    'observable_file_hash': 2
+}
+
+TLP = {
+    'WHITE': 0,
+    'GREEN': 1,
+    'AMBER': 2,
+    'RED': 3,
+}
+
+CONFIDENCE = {
+    'LOW': 0,
+    'MEDIUM': 1,
+    'HIGH': 2,
+}
+
+
+def get_observable_type(reported_type):
+    reported_type = reported_type.lower()
+    if 'ip ' in reported_type:
+        return 0
+    if 'domain' in reported_type:
+        return 1
+    if 'url' in reported_type:
+        return 2
+    if 'file' in reported_type:
+        return 4
+    return False
+
+
+def get_observable_value(obs_type, row):
+    if obs_type == 4:
+        return row[COLUMNS['observable_file_hash']]
+    return row[COLUMNS['observable_value']]
+
+
+def get_hash_type(file_hash):
+    hash_len = len(file_hash)
+    if hash_len == 32: #MD5
+        return 0
+    if hash_len == 40: #SHA1
+        return 1
+    if hash_len == 56: #SHA224
+        return 2
+    if hash_len == 64: #SHA256
+        return 3
+    return False
```

Comment:
This could also be achieved with a table:
```py
def get_hash_type(file_hash, default=False):
    HASH_TYPE_BY_LENGTH = {
        32: 0,  # MD5
        40: 1,  # SHA-1
        56: 2,  # SHA-224
        64: 3,  # SHA-256
    }
    hash_len = len(file_hash)
    return HASH_TYPE_BY_LENGTH.get(hash_len, default)
```